code/src/components/TodoItem.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import { todos } from '../reducers/todos.js'
+import { useSelector, useDispatch } from 'react-redux'
+// import { TimeStamp } from './TimeStamp'
+import { todoItem } from './todoItem.css'
+import styled from 'styled-components/macro'
+import { CustomCheckbox } from 'lib/CustomCheckbox'
+
+export const TodoItem = ({ itemIndex }) => {
+  const item = useSelector(store => store.todos.list.items[itemIndex])
+
+  const dispatch = useDispatch()
+
+  const onRemoveClicked = event => {
+    dispatch(
+      todos.actions.removeTodo({
+        itemIndex: itemIndex
+      })
+    )
+  }
+
+  const handleOnChange = event => {
+    dispatch(
+      todos.actions.setDone({
+        itemIndex: itemIndex,
+        done: !item.done
+      }))
```

Comment:
It would be better to use a unique id instead of the index to deal with the done state. There's an issue now that happens if you write some todos, check the first one and then remove it, the second one will "jump up" to index 0 and become checked.